app/code/Magento/Catalog/Ui/DataProvider/Product/ProductDataProvider.php

```diff
@@ -52,12 +60,14 @@ public function __construct(
         array $addFieldStrategies = [],
         array $addFilterStrategies = [],
         array $meta = [],
-        array $data = []
+        array $data = [],
+        PoolInterface $modifiersPool = null
     ) {
         parent::__construct($name, $primaryFieldName, $requestFieldName, $meta, $data);
         $this->collection = $collectionFactory->create();
         $this->addFieldStrategies = $addFieldStrategies;
         $this->addFilterStrategies = $addFilterStrategies;
+        $this->modifiersPool = $modifiersPool ?: ObjectManager::getInstance()->get(PoolInterface::class);
```

Comment:
Problem is next: somebody extended `ProductDataProvider` and called the parent constructor in own child constructor 
If you add one more required argument to parent class it will be lead to a fatal error in child constructor